rush-plugins/rush-azure-storage-build-cache-plugin/src/AdoCodespacesAuthCredential.ts

```diff
@@ -0,0 +1,28 @@
+// Copyright (c) Microsoft Corporation. All rights reserved. Licensed under the MIT license.
+// See LICENSE in the project root for license information.
+import type { AccessToken, GetTokenOptions } from '@azure/identity';
+
+export class AdoCodespacesAuthCredential {
+  // eslint-disable-next-line @rushstack/no-new-null
+  public async getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | null> {
```

Comment:
```suggestion
  public async getToken(scopes: string | string[], options?: GetTokenOptions): Promise<AccessToken | undefined> {
```

Reply:
What should happen if you try to run this on a non-codespaces machine?

Reply:
If it runs on a in a non-codespaces environment or does not have the ado-codespaces-auth vscode extension installed then the `azure-auth-helper` bin won't exist and the exec will throw.

I could catch that and throw a error that says this is meant to run on a codespace with that extension?

Reply:
We could do a try/catch for AdoCodespacesAuth and fallback to DeviceCode?

Reply:
The typing needs to have null to be compatible with `TokenCredential` from `@azure/identitiy`

Reply:
perhaps a fallback chain? `AdoCodespacesAuth -> InteractiveBrowser -> DeviceCode`